hawkular-inventory-rest-api/src/main/rest-doc/base.adoc

```diff
@@ -16,3 +16,362 @@ When you send JSON data with a `POST` or `PUT` request, you _must_ add the follo
+
+NOTE: In the future, there is a plan to support full Gremlin-based access to inventory.
+
+The URL traversal of the inventory is an extension of the format of canonical path. E.g.
+
+  /hawkular/inventory/traversal/f;feedId/r;resourceId
+
+is a valid URL traversal, because `/f;feedId/r;resourceId` is a valid (tenantless) canonical path.
+
+To be able to formulate efficient traversals, it is good to understand the inventory organization. Please consult
+the http://www.hawkular.org/docs/components/inventory/index.html#inventory-organization[documentation] for its details.
+
+Now to a more formal definition of the URL traversal (for the technically inclined, the traversal is described using an
+http://github.com/hakwular/hawkular-inventory/hawkular-inventory-rest-api/src/main/antlr4/org/hawkular/inventory/rest/HawkularInventoryGetUri.g4[Antlr4 grammar]).
+
+===== Filtering Entities
+
+Each path segment in the URL traversal expresses a filter on the "current" set of entities. The filtering is performed
+within a single URL path segment using a notation similar to HTTP matrix parameters.
+
+Given an example traversal `/hawkular/inventory/traversal/f;feedId;name=My%20Feed/r;resourceId` we'll be talking about
+a single path segment, e.g. `f;feedId;name=My%20Feed` or `r;resourceId`.
+
+Given the example `f;feedId;name=My%20Feed` you can spot two things:
+
+* multiple filters can be used, separated by a semi-colon,
+* within a single path segment everything must be percent-encoded (i.e. above we're looking for a feed with id `feedId`
+and name `My Feed`). Again this is in accordance with canonical path format which requires percent-encoding for IDs,
+too.
+
+//just splitting the above list from the below one
+Canonical path segment as filter::
+`r;resourceId` or `mt;AwesomeMetricType` are valid filters. The (one or two) letters perceding the first semicolon
+specify the type of the sought-after entities (for the full list, check
+http://hawkular.org/docs/components/inventory/index.html#basic-principles[canonical path format description]).
+
+Other Filters::
+** `type` -- only allows entities of the specified type. The type can be specified either as the segment type as in
+the canonical path segments (`type=d`) or by their full name (no spaces, first letter lower-case, camel-case afterwards,
+e.g. `type=resource`, `type=resourceType`, etc.). Multiple occurences of the `type` filter are or-ed together, i.e.
+the sought-after entities must have one of the types specified.
+** `id` -- only allows entities with the specified id. Multiple occurences of the filter are or-ed together, i.e. the
+entities must have only one of the provided ids.
+** `name` -- only allows entities with the specified name. Again, multiple occurences are or-ed together.
+** `cp` -- only allows entities with the given canonical path. Note that you have to percent-encode the value so that
+it can appear as a filter, e.g. `.../cp=%2Ft%3BtenantId%2Ff%3BfeedId/...`
```

Comment:
well, something has to give ;) It's either the ability to do ad-hoc traversals or some level of inconvenience with certain constructs.